modules/pubmaticBidAdapter.js

```diff
@@ -63,7 +63,21 @@ const converter = ortbConverter({
     ttl: DEFAULT_TTL
   },
   imp(buildImp, bidRequest, context) {
-    const { kadfloor, currency, adSlot = '', deals, dctr, pmzoneid, hashedKey } = bidRequest.params;
+    const { kadfloor, currency, adSlot = '', deals, dctr: origDctr, pmzoneid, hashedKey } = bidRequest.params;
+    let dctr = origDctr;
+
+    // Additive dctr logic: append IAS brand safety data if present
+    const ortb2Data = bidRequest?.ortb2 || {};
+    const iasBrandSafety = ortb2Data?.site?.ext?.data?.['ias-brand-safety'];
```

Comment:
const ortb2Data = bidRequest?.ortb2 || {};
const iasBrandSafety = ortb2Data?.site?.ext?.data?.['ias-brand-safety'];
can be replaced with below line, to make it more cleaner
const iasBrandSafety = bidRequest?.ortb2?.site?.ext?.data?.['ias-brand-safety'] || {};